docs/UserGuide.md

```diff
@@ -181,18 +181,32 @@ Clears all entries from the address book.
 
 Format: `clear`
 
-### Filtering persons by tags : `filter`
+### Filtering persons : `filter`
 
-Filters all persons whose tags contain any of the specified keywords. The filtering is case-insensitive.
+Filters and lists all persons in address book whose fields (name, role, email, phone, address) match any of the
+specified keywords (case-insensitive).
 
-Format: `filter t/KEYWORD t/[MORE_KEYWORDS]...`
+Format: `filter [n/NAME] [r/ROLE] [e/EMAIL] [p/PHONE] [a/ADDRESS]`
 
-* Filters all persons whose tags contain any of the specified keywords.
-* The search is case-insensitive. e.g `t/friend` will match `t/Friend`
-* The order of the keywords does not matter. e.g. `t/friend t/colleague` will match `t/colleague t/friend`
-* Only full words will be matched e.g. `t/col` will not match `t/colleague`
+* At least one field must be provided.
+* Parameters can be in any order.
+* Each field can only contain single word keywords.
```

Comment:
I think can remove the line about single word keywords, and specify below instead, since address field can have more than 1 word keyword.